### ISSUE TITLE
Consider using fsspec for file access

### ISSUE THREAD

**Issue by kosack** [cta-observatory/ctapipe#2145]
**Please describe the use case that requires this feature.**
Since astropy now optionally uses the fsspec library for open local and remote FITS files, and this library is also used by pandas and many others, it might be useful to replace parts of ctapipe's  URL functionality (`ctapipe.utils.download` and `ctapipe.utils.download_cached`) with it.

It supports opening files from many filesystem sources (see list below) and using many compression methods, and supports caching as we use now.

**Additional context**

e.g.:
```py3
of = fsspec.open("github://cta-observatory:ctapipe@master/environment.yml")
with of as f:
    print(f.read())
```

Available support:
```py3
In [4]: fsspec.available_compressions()
Out[4]: [None, 'zip', 'bz2', 'gzip', 'lzma', 'xz', 'lz4', 'zstd']

In [5]: fsspec.available_protocols()
Out[5]:
['file',
 'memory',
 'dropbox',
 'http',
 'https',
 'zip',
 'tar',
 'gcs',
 'gs',
 'gdrive',
 'sftp',
 'ssh',
 'ftp',
 'hdfs',
 'arrow_hdfs',
 'webhdfs',
 's3',
 's3a',
 'wandb',
 'oci',
 'asynclocal',
 'adl',
 'abfs',
 'az',
 'cached',
 'blockcache',
 'filecache',
 'simplecache',
 'dask',
 'dbfs',
 'github',
 'git',
 'smb',
 'jupyter',
 'jlab',
 'libarchive',
 'reference',
 'generic',
 'oss',
 'webdav',
 'dvc',
 'root']
```

**Comment by kosack** [comment]
downside of course would be one more dependency, which maybe is not really needed (file access may only be via local files for the most part).

**Comment by HealthyPear** [comment]
> I don't see how that directly relates to input to ctapipe. Could you ofer some clarification?

Yes, probably this issue is about something yet different. Seeing "SSH" as one of the available protocols I thought it might have to do also with the actual connection - am I wrong?

As a "corollary", I wanted to understand if `ctapipe.utils` plans to support SSH tunneling to e.g. get test data from a server - I have the feeling that it works only via HTTPS.

**Comment by maxnoe** [comment]
> (which we currently do not I think)

We just pass the URL to `requests`, so it will support non-standard ports

**Comment by maxnoe** [comment]
I think we probably should do this, as this will also enable us to avoid copying files to worker nodes on the grid if we can directly read from protocols like root:// or others that are supported by the storage elements / DIRAC.

Dirac has the option to either download files for the job and put them into the current directory or just to provide  a url.
